packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1861,6 +1902,13 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             type: envelope.contents.type,
         };
 
+        // Only collect signal telemetry for messages sent by the current client.
+        if (message.clientId === this.clientId && envelope.clientSignalSequenceNumber !== undefined) {
+            if (envelope.clientSignalSequenceNumber % this.defaultTelemetrySignalSampleCount === 1 && this.connected) {
```

Comment:
I'd not rely here on having logic in sync with createNewSignalEnvelope (re comparison to 1).
I'd actually create a new variable trackingSignalSequenceNumber and would set it in createNewSignalEnvelope to signify what signal we are tracking.

Then here, you can do better comparison, but also report error to telemetry if you see envelope.clientSignalSequenceNumber > trackingSignalSequenceNumber  (in both success and this failure mode we need to reset state).

I'd also not start tracking any new signals in createNewSignalEnvelope if trackingSignalSequenceNumber  !== undefined (maybe client sent 1K signals in a row).

Reply:
I can - the only con is that I would need to add a check to every signal instead of every 100ish 
Something in the lines of: 

            if (this._perfSignalData.trackingSignalSequenceNumber !== undefined &&
                envelope.clientSignalSequenceNumber > this._perfSignalData.trackingSignalSequenceNumber) {
                this._perfSignalData.signalsLost++;
                this._perfSignalData.trackingSignalSequenceNumber = undefined;
            } else if (envelope.clientSignalSequenceNumber % this.defaultTelemetrySignalSampleCount === 1) {
                this.sendSignalTelemetryEvent(envelope.clientSignalSequenceNumber);
                this._perfSignalData.trackingSignalSequenceNumber = undefined;
            }

Reply:
You already check every signal in processSignal(), right? It same thing, no?